### PR TITLE
Move from termion to termwiz for STDIN handling

### PR DESCRIPTION
**Pull request by imsnif**
This change lays down some groundwork for:
1. windows support (termwiz is cross platform)
2. multiple modifier key support
3. allow navigation with alt+arrow keys
4. much more!

### CODE REVIEW

**Comment by imsnif** [comment]
@tlinford - I'm still working on the flakiness of the e2e tests that this introduced, but would be happy for your input as I'm doing so (time allowing, ofc).

I had to change the `normal_mouse_report` functions in cases.rs of the e2e tests to make this work, and I'm honestly not sure what the previous function was doing. If you could also take a look at that to make sure I didn't mess it up, that would be grand.

**Comment by tlinford** [comment]
Hey @imsnif,
I played around with this a bit last evening, seems to be working great!

I left a comment above about the one thing that seems to be missing, which is the mouse event repeater that makes scrolling up/down while selecting work.

Seems like termwiz is doing a lot of things right, the one thing that has me a bit concerned is the mouse events - but I think the solution you made with `InputHandler.holding_mouse` should be about equivalent to what we have now in termion, which is fine I guess but can get a bit wonky (try for example holding one mouse button and clicking with another).

> I had to change the `normal_mouse_report` functions in cases.rs of the e2e tests to make this work, and I'm honestly not sure what the previous function was doing. If you could also take a look at that to make sure I didn't mess it up, that would be grand.

This is encoding mouse events using the "Normal" (1000) encoding: https://invisible-island.net/xterm/ctlseqs/ctlseqs.html#h3-Normal-tracking-mode. Not sure why it breaks with termwiz, but might be worth investigating a little further, because if it doesn't work we should probably not try and enable it here either: https://github.com/zellij-org/zellij/blob/23e0315f8a0ecfac28303833932df6fe8ae1f564/zellij-client/src/os_input_output.rs#L23.

File: src/tests/e2e/cases.rs [hunk 1]
```diff
@@ -58,15 +58,16 @@ pub const BRACKETED_PASTE_START: [u8; 6] = [27, 91, 50, 48, 48, 126]; // \u{1b}[
 pub const BRACKETED_PASTE_END: [u8; 6] = [27, 91, 50, 48, 49, 126]; // \u{1b}[201
 pub const SLEEP: [u8; 0] = [];
 
-// simplified, slighty adapted version of alacritty mouse reporting code
 pub fn normal_mouse_report(position: Position, button: u8) -> Vec<u8> {
```
Comment (tlinford) [inline src/tests/e2e/cases.rs:61]: nitpicking, but should probably call this sgr_mouse_report now

File: zellij-client/src/stdin_handler.rs [hunk 2]
```diff
@@ -1,81 +1,27 @@
...
-        }
-
-        if let termion::event::Event::Mouse(me) = key_event {
-            let mouse_event = zellij_utils::input::mouse::MouseEvent::from(me);
-            if let MouseEvent::Hold(_) = mouse_event {
-                // as long as the user is holding the mouse down (no other stdin, eg.
-                // MouseRelease) we need to keep sending this instruction to the app,
-                // because the app itself doesn't have an event loop in the proper
-                // place
-                let mut poller = os_input.stdin_poller();
```
[the remark this answers is not in the document]
Comment (imsnif) [reply zellij-client/src/stdin_handler.rs:42]: Readded, thanks!

**Pull request merged**